agent/consul/connect_ca_endpoint.go

```diff
@@ -163,6 +163,22 @@ func (s *ConnectCA) ConfigurationSet(
 		}
 	}()
 
+	// If this is a secondary, check if the intermediate needs to be regenerated.
+	if s.srv.config.Datacenter != s.srv.config.PrimaryDatacenter {
+		// Attempt to take the CA lock in order to update the config.
+		if err := s.srv.caManager.setState(CAStateReconfig); err != nil {
+			return err
+		}
+		defer s.srv.caManager.setReady()
```

Comment:
So if the init function call fails, you'll still revert to READY. Is that correct? Is there any other cleanup required?

Reply:
There shouldn't be any cleanup required since nothing is persisted until the very end, i.e. `persistNewRootAndConfig`, so if that succeeds then there's noithing to error after it, and if something before that fails then there's no changes to worry about.